### PR TITLE
Write per-virus per-sample fasta in proteins-to-viruses.py

### LINKED ISSUE #465: Have proteins-to-viruses.py write out per-virus per-sample FASTA
These files should probably be added to the `out` directory.

### PR DESCRIPTION
**Pull request by terrycojones**
Fixes #465.

### CODE REVIEW

File: dark/proteins.py [hunk 1]
```diff
@@ -4,7 +4,52 @@
...
 from dark.html import NCBISequenceLinkURL
+from dark.reads import Reads
+
+
+class VirusSampleFASTA(object):
+    """
+    Maintain a cache of virus/sample FASTA file names, creating the FASTA
+    on demand.
+    """
+    def __init__(self, proteinGrouper):
```
Comment (bamueh) [inline dark/proteins.py]: You don't have an explanation for `proteinGrouper`.

File: dark/proteins.py [hunk 2]
```diff
@@ -286,7 +336,7 @@ def toHTML(self):
 
         # Write all samples (with viruses (with proteins)).
         append('<h1>Samples by virus</h1>')
-        for sampleName in sorted(sampleNames):
+        for sampleName in sampleNames:
```
Comment (bamueh) [inline dark/proteins.py]: You don't need the newline.

File: test/test_proteins.py [hunk 3]
```diff
@@ -264,8 +271,9 @@ def testOneLineInEachOfTwoFilesSameVirusTitle(self):
 
     def testOneLineInEachOfTwoFilesDifferentViruses(self):
         """
-        If a protein grouper is given two files, each with one line from the
-        different viruses, its virusTitles dict must be as expected.
+        If a protein grouper is given two files intwo different directories,
```
Comment (bamueh) [inline test/test_proteins.py]: s/intwo/in two

**Comment by bamueh** [comment]
OK, just the trivial comments.

File: dark/proteins.py (hunk 2 above)
Comment (terrycojones) [inline dark/proteins.py]: OK.

File: dark/proteins.py (hunk 1 above)
Comment (terrycojones) [inline dark/proteins.py]: Fixed.

File: test/test_proteins.py (hunk 3 above)
Comment (terrycojones) [inline test/test_proteins.py]: Fixed.

**Pull request merged**